index.js

```diff
@@ -104,21 +107,48 @@ export default e => {
       ).premultiply(app.matrixWorld).decompose(npcPlayer.position, npcPlayer.quaternion, localVector3);
       npcPlayer.updateMatrixWorld(); */
 
-      // window.npcPlayer = npcPlayer;
+      window.npcPlayer = npcPlayer; // test
 
-      if (target) {
-        const v = new THREE.Vector3().setFromMatrixPosition(target.matrixWorld)
+      if (target && npcFarawayLocalPlayer()) {
+        const isInitial = !pathFinder.destVoxel
+
+        if (isInitial || localPlayerFarawayPrevDest()) {
+          // console.log('localPlayerFarawayPrevDest')
+
+          // Don't allowNearest here, otherwise will get inaccurate result when accurate result already exists.
+          const isFound = pathFinder.getPath(npcPlayer.position, localPlayer.position, false);
+          // if (pathFinder.startVoxel) target = pathFinder.startVoxel;
+          if (isFound) target = pathFinder.waypointResult[0];
+        }
+
+        if (pathFinder.waypointResult.length > 0 && npcReachedDest()) {
```

Comment:
Why not just return the object or `null`? This is a weird way to access it. I realize it is a cache but I think it is ok to return the cache, even if it is destroyed the next frame.

Reply:
Ok, I'll change.

Reply:
Changed.
https://github.com/webaverse/app/pull/2298/commits/87f2eac9792ef350ca256b9198686d5a672e0b3e
https://github.com/webaverse/npc/pull/2/commits/e157f7033647e312b6b7b4feb66e9c3f3f21a225